components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/RegistryContentResource.java

```diff
@@ -0,0 +1,441 @@
+                        axis2MessageContext, INTERNAL_SERVER_ERROR);
+                Utils.setJsonPayLoad(axis2MessageContext, jsonBody);
+            }
+        } else {
+            JSONObject jsonBody = Utils.createJsonError("Error occurred while reading the requested file",
+                    axis2MessageContext, INTERNAL_SERVER_ERROR);
+            Utils.setJsonPayLoad(axis2MessageContext, jsonBody);
+        }
+    }
+
+    /**
+     * This method fetches content from the registry.
+     *
+     * @param path  Registry path
+     * @return      DataHandler object with registry content
+     */
+    private DataHandler downloadRegistryFiles(String path) {
+
+        ByteArrayDataSource bytArrayDS;
+        String carbonHomePath = Utils.getCarbonHome();
+        String registryPath = formatPath(carbonHomePath + File.separator + path);
+        File file = new File(registryPath);
+        if (file.exists() && !file.isDirectory()) {
+            try (InputStream is = new BufferedInputStream(new FileInputStream(registryPath))) {
+                bytArrayDS = new ByteArrayDataSource(is, "text/xml");
+                return new DataHandler(bytArrayDS);
+            } catch (FileNotFoundException e) {
+                LOG.error("Could not find the requested file : " + path + " in : " + registryPath, e);
+                return null;
+            } catch (IOException e) {
+                LOG.error("Error occurred while reading the file : " + path + " in : " + registryPath, e);
+                return null;
+            }
+        } else {
+            LOG.error("Could not find the requested file : " + path + " in : " + registryPath);
+            return null;
+        }
+    }
+
+    /**
+     * This method handles POST.
+     *
+     * @param messageContext      Synapse message context
+     * @param axis2MessageContext AXIS2 message context
+     */
+    private void handlePost(MessageContext messageContext,
+            org.apache.axis2.context.MessageContext axis2MessageContext) {
+
+        String registryPath = Utils.getQueryParameter(messageContext, REGISTRY_PATH);
+        String validatedPath;
+        JSONObject jsonBody;
+        axis2MessageContext.removeProperty(Constants.NO_ENTITY_BODY);
+        if (Objects.nonNull(registryPath)) {
+            validatedPath = validatePath(registryPath, axis2MessageContext);
+            if (Objects.nonNull(validatedPath)) {
+                String pathWithPrefix = getRegistryPathPrefix(validatedPath);
+                if (isRegistryExist(validatedPath)) {
+                    jsonBody = Utils.createJsonError("Registry already exists. Can not POST an existing registry",
+                            axis2MessageContext, BAD_REQUEST);
+                } else if (Objects.nonNull(pathWithPrefix)) {
+                    jsonBody = postRegistryArtifact(messageContext, axis2MessageContext, pathWithPrefix);
```

Comment:
Let's add audit logs for MI management modify operations. Check the following sample:

https://github.com/wso2/micro-integrator/blob/master/components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/ProxyServiceResource.java#L258

Reply:
I will add audit logs

Reply:
Added audit logs